test_aodndata/argo/test_argoHandler.py

```diff
@@ -19,6 +23,28 @@ def test_rsync_manifest_file(self):
         handler.resolve_params = {'relative_path_root': os.path.join(TEST_ROOT, 'Argo/dac')}
         handler.run()
 
+        f = handler.file_collection[0]
+        self.assertEqual(f.check_type, PipelineFileCheckType.FORMAT_CHECK)
+        self.assertEqual(f.publish_type, PipelineFilePublishType.HARVEST_UPLOAD)
+        self.assertEqual(f.dest_path, 'IMOS/Argo/dac/csiro/1901119/profiles/D1901119_001.nc')
+        self.assertTrue(f.is_stored)
+
+    def test_missing_file_rsync_manifest_file(self):
+        # check file in manifest but not in filesystem
+        handler = self.handler_class(MISSING_FILE_RSYNC_MANIFEST_FILE)
+        handler.resolve_params = {'relative_path_root': os.path.join(TEST_ROOT, 'Argo/dac')}
+        handler.run()
+
+        self.assertIsInstance(handler.error, MissingFileError)
+
+    def test_non_nc_rsync_manifest_file(self):
+        # check file not finishing with nc extension is not pusblished
+        handler = self.handler_class(NON_NC_RSYNC_MANIFEST_FILE)
+        handler.resolve_params = {'relative_path_root': os.path.join(TEST_ROOT, 'Argo/dac')}
+        handler.run()
```

Comment:
According to Leigh, it's better to use the run_handler class 
`
handler = self.run_handler()
`

Reply:
Yes! The reason for this is that the handler does *not* raise an exception in case of an error, so it can hide bugs and errors if the test is not explicitly looking for it.

run_handler calls the run method and then checks the "handler.error" attribute is None, otherwise it fails the test.

https://aodn.github.io/python-aodncore/_modules/aodncore/testlib/handlertest.html#HandlerTestCase.run_handler

Reply:
@lwgordonimos I have modified the code to add some preprocess. also made the unittests more explicit

Reply:
"test_non_nc_rsync_manifest_file" doesn't have an assertion for handler.error? This should *always* assert either None if no error expected, or the specific Exception class if one is expected.